### PR TITLE
doc(sdk): Rephrase the Quick refreshing Section

### PR DESCRIPTION
**Pull request by Hywan**
With https://github.com/matrix-org/matrix-rust-sdk/pull/1601, it's safer to cancel a stream at any time, as any `Future` cancellation. This documentation section of `SlidingSync` is no longer relevant as is, and can be rephrased.

Thoughts?

### CODE REVIEW

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 1]
```diff
@@ -323,94 +323,6 @@
...
-//! waiting for a `stream.next().await` call before the next request is sent.
-//! The [specification][MSC] on long polling also states, however, that if an
-//! new request is found coming in, the previous one shall be sent out. In
-//! practice that means one can just start a new stream and the old connection
-//! will return immediately — with a proper response though. One just needs to
-//! make sure to not call that stream any further. Additionally, as both
-//! requests are sent with the same positional argument, the server might
-//! respond with data, the client has already processed. This isn't a problem,
-//! the [`SlidingSync`][] will only process new data and skip the processing
-//! even across restarts.
```
Comment (jplatte) [inline crates/matrix-sdk/src/sliding_sync/mod.rs:343]: Maybe some of this is still relevant? Agree that the code example is probably not all that helpful, especially now that we've changed the cancellation situation.

Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/mod.rs:343]: Well, there is nothing special to do when cancelling the result of `SlidingSync::stream`. Let's see tomorrow.

Comment (Hywan) [reply crates/matrix-sdk/src/sliding_sync/mod.rs]: What about now :-)?

File: crates/matrix-sdk/src/sliding_sync/mod.rs [hunk 2]
```diff
@@ -342,74 +343,16 @@
...
-//!
-//! # anyhow::Ok(())
-//! # });
-//! ```
-//!
+//! To support this, in practice, one can spawn a `Future` that runs
+//! [`SlidingStream::stream`]. The spawned `Future` can be cancelled safely. If
+//! the client was waiting on a response, it's cancelled without any issue. If
+//! the client was receiving a response, or a response was just received, the
+//! response will be fully handled by `SlidingSync`. This _response is always
```
Comment (jplatte) [inline crates/matrix-sdk/src/sliding_sync/mod.rs:350]:
```suggestion
//! the client was waiting on a response, it's cancelled without any issue. If
//! a response was just received, it
//! will be fully handled by `SlidingSync`. This _response is always
```

**Review by jplatte — APPROVED** [review]

**Pull request merged**